Content.Server/GameTicking/Rules/TraitorRuleSystem.cs

```diff
@@ -221,60 +219,87 @@ public bool MakeTraitor(IPlayerSession traitor)
             _sawmill.Info("Failed getting mind for picked traitor.");
             return false;
         }
-
         if (mind.OwnedEntity is not { } entity)
         {
             Logger.ErrorS("preset", "Mind picked for traitor did not have an attached entity.");
             return false;
         }
 
-        // creadth: we need to create uplink for the antag.
-        // PDA should be in place already
-        DebugTools.AssertNotNull(mind.OwnedEntity);
-
+        // Calculate the amount of currency on the uplink.
         var startingBalance = _cfg.GetCVar(CCVars.TraitorStartingBalance);
-
         if (mind.CurrentJob != null)
             startingBalance = Math.Max(startingBalance - mind.CurrentJob.Prototype.AntagAdvantage, 0);
 
+        // creadth: we need to create uplink for the antag.
+        // PDA should be in place already
         var pda = _uplink.FindUplinkTarget(mind.OwnedEntity!.Value);
         if (pda == null || !_uplink.AddUplink(mind.OwnedEntity.Value, startingBalance))
             return false;
 
+        // Add the ringtone uplink and get its code for greeting
+        var code = EnsureComp<RingerUplinkComponent>(pda.Value).Code;
 
-        // add the ringtone uplink and get its code for greeting
-        var code = AddComp<RingerUplinkComponent>(pda.Value).Code;
-
+        // Prepare antagonist role
         var antagPrototype = _prototypeManager.Index<AntagPrototype>(traitorRule.TraitorPrototypeId);
         var traitorRole = new TraitorRole(mind, antagPrototype);
+
+        // Give traitors their codewords and uplink code to keep in their character info menu
+        traitorRole.Mind.Briefing = string.Format(
+            "{0}\n{1}",
+            Loc.GetString("traitor-role-codewords-short", ("codewords", string.Join(", ", traitorRule.Codewords))),
+            Loc.GetString("traitor-role-uplink-code-short", ("code", string.Join("", code))));
+
+        // Assign traitor roles
         mind.AddRole(traitorRole);
+        SendTraitorBriefing(mind, traitorRule.Codewords, code);
         traitorRule.Traitors.Add(traitorRole);
-        traitorRole.GreetTraitor(traitorRule.Codewords, code);
 
+        if (mind.TryGetSession(out var session))
+        {
+            // Notificate player about new role assignment
+            _audio.PlayGlobal(
+                traitorRule.GreetSoundNotification,
+                session,
+                traitorRule.GreetSoundNotification.Params);
```

Comment:
unnecesary argument since PlayGlobal *should* just get the GreetSoundNotification params if none are given